doc/build-helpers/testers.chapter.md

```diff
@@ -30,38 +30,55 @@ meta.pkgConfigModules = [ "libfoo" ];
 
 ## `testVersion` {#tester-testVersion}
 
-Checks the command output contains the specified version
+Checks that the output from running a command contains the specified version string in it as a whole word.
 
 Although simplistic, this test assures that the main program
 can run. While there's no substitute for a real test case,
 it does catch dynamic linking errors and such. It also provides
 some protection against accidentally building the wrong version,
 for example when using an 'old' hash in a fixed-output derivation.
 
-Examples:
+By default, the command to be run will be inferred from the given `package` attribute:
+it will check `meta.mainProgram` first, and fall back to `pname` or `name`.
+The default argument to the command is `--version`,
+and the version to be checked will be inferred from the given `package` attribute as well.
+
+:::{.example #ex-testversion-hello}
+
+# Using all the default values
```

Comment:
```suggestion
# Check a program version using default values
```

Examples explain themselves on their own, as they will be compiled in the appendix where the context is not available.